linode/manager
# refactor: [M3-6625] - Make changes to facilitate Cloud Manager region test pipeline

## Description 📝
This makes a few small changes to facilitate our automated region testing plans.

- **Docker changes:**
  - By default, Cypress runner containers will now only call `yarn cy:ci`, and any extra parameters can be passed by overriding the command via Docker Compose
  - Passing a couple environment variables that were previously unused or hardcoded: `CYPRESS_BASE_URL`, `CY_TEST_SUITE`, `CY_TEST_REGION`, `CY_TEST_USER_REPORT`.

- **Cypress changes:**
  - Further improved handling of `MANAGER_OAUTH` environment variable when tests begin. In addition to reporting if `MANAGER_OAUTH` is not defined, Cypress will now make a request to Linode APIv4 to confirm that the OAuth token is valid.
  - Added a new environment variable, `CY_TEST_USER_REPORT`. When it is set, the username of the test account being used will be reported in the Cypress console output.

## How to test 🧪
~I'm primarily concerned with making sure these changes do not impact our existing test suite, and thankfully we can lean on the automated run for that.~ On second thought, I'm pretty sure that this PR will have to be merged after the pipeline PR and that the automated run will fail.

To confirm the other changes made by this PR:

* Confirm `MANAGER_OAUTH` handling improvements:
  1. With your valid OAuth token defined in your `.env` file, run `yarn cy:run` to confirm that Cypress reads your token as expected (confirm that the existing behavior has not changed). You can exit as soon as a test passes.
  2. Run `MANAGER_OAUTH='you-can-put-any-invalid-value-here' yarn cy:run` and confirm that Cypress exits quickly, and that it reports that it failed to make a request to the API (you should also see a 401 response noted).
* Confirm `CY_TEST_USER_REPORT` environment variable handling:
  1. Run `CY_TEST_USER_REPORT=1 yarn cy:run` and confirm that your test account's username is shown near the beginning of Cypress's console output
  2. Run `yarn cy:run` and confirm that your test account username is not shown near the beginning of Cypress's console output
